### DISCUSSION TITLE
Missing certain CONCOURSE_GARDEN_* env vars for Containerd?

### DISCUSSION THREAD

**Discussion by mvdkleijn** [concourse/concourse#6874] [Help & Support]
Hi,

In our setup we use (among others) these environment variables on the workers:
```
CONCOURSE_GARDEN_DOCKER_REGISTRY
CONCOURSE_GARDEN_DEPOT
CONCOURSE_GARDEN_NETWORK_POOL
CONCOURSE_GARDEN_REQUEST_TIMEOUT
CONCOURSE_GARDEN_ADDITIONAL_DNS_SERVER
CONCOURSE_GARDEN_DNS_SERVER
CONCOURSE_GARDEN_DNS_PROXY_ENABLE
```
In trying to re-configure the workers to use containerd as mentioned in the release notes of v7.x, some of these seem to have no equivalent env variable for containerd.

I guess my question is: is that correct and is containerd not ready for use yet or am I missing something?

Sincerely, Martijn

**Comment by mvdkleijn** [comment]
Maybe I should turn this into an issue instead of a discussion / question... :)

**Comment by taylorsilva** [comment]
@muntac pinging you; is this a documentation issue or are there still some gaps between guardian and containerd here?

**Reply by muntac** [reply]
We don't support `CONCOURSE_GARDEN_DOCKER_REGISTRY` and `CONCOURSE_GARDEN_DEPOT` in `containerd`.

@mvdkleijn I'm not familiar with these features. Could you please describe how you're using them?

**Reply by mvdkleijn** [reply]
Hi @muntac Sorry for the late reply, it's a little hectic here.

`CONCOURSE_GARDEN_DEPOT` equates to this option from `gdn server --help`:
`--depot=                             Directory in which to store container data. (default: /var/run/gdn/depot)`

`CONCOURSE_GARDEN_DOCKER_REGISTRY` equates to this option from `gdn server --help`:
`--docker-registry=                   Docker registry API endpoint. (default: registry-1.docker.io)`

These are configured according to these docs: https://concourse-ci.org/concourse-worker.html#configuring-gdn-server

The first we use to store the data in a specific location.
The second we use because we have an internal registry that also acts as a proxy for Docker hub.

I hope that's clear, if you have any more questions, please let me know.

**Reply by aoldershaw** [reply]
@mvdkleijn I'm looking at the Guardian code, and I see the `--docker-registry` flag defined, but it doesn't look like it gets used anywhere. Either way, I'd be pretty surprised if it was relevant for use in Concourse in the first place, given that it's Concourse that does all the image pulling - not Guardian. Concourse always provides the `RootFSPath` under the `raw://` scheme, never `docker://`
